src/main/java/randoop/condition/PreconditionOutcomeTable.java

```diff
@@ -0,0 +1,138 @@
+import randoop.test.PostConditionCheckGenerator;
+import randoop.test.TestCheckGenerator;
+
+/**
+ * Records the outcome of checking all of the preconditions for a method. Each table entry records:
+ *
+ * <p>TODO: clarify the following, or merge it into other documentation.
+ *
+ * <ol>
+ *   <li>Whether the preconditions of the specification fail or are satisfied. The preconditions
+ *       fail if the Boolean expression of any precondition in the specification is false.
+ *       Otherwise, the preconditions are satisfied. See {@link
+ *       OperationConditions#checkPreconditions(java.lang.Object[])}.
+ *   <li>A set of expected exceptions. The exception is expected because its guard was satisfied.
+ *       Evaluate the guard of each throws-condition, and for each one satisfied, add the exception
+ *       to the set of expected exceptions. (There will be one set per specification.) See {@link
+ *       OperationConditions#checkThrowsPreconditions(java.lang.Object[])}.
+ *   <li>The expected postcondition, if any. If the preconditions are satisfied, test the guards of
+ *       the normal postconditions of the specification in order, and save the property for the
+ *       first guard satisfied, if there is one. See {@link
+ *       OperationConditions#checkPostconditionGuards(java.lang.Object[])}.
+ * </ol>
+ *
+ * <p>TODO: Merge the following in as well
+ *
+ * <ul>
+ *   <li>Whether the preconditions fail or are satisfied, given the observed values
+ *   <li>The set of expected exceptions, given the observed values (that is, which
+ *       throws-conditions' guards are satisfied)
+ *   <li>The expected postcondition, given the observed values (that is, which return-clauses'
+ *       guards are satisfied)
+ * </ul>
+ */
+public class PreconditionOutcomeTable {
+
+  /** Indicates whether this table is empty. */
+  private boolean isEmpty = true;
+
+  /** Indicates whether this table has a valid entry. */
+  private boolean hasValid = false;
+
+  /** The list of throws clauses for this table. */
+  private final List<Set<ThrowsClause>> exceptionSets;
+
+  /** The list of post-conditions for this table. */
+  private final List<PostCondition> postConditions;
+
+  /** Creates an empty {@link PreconditionOutcomeTable}. */
+  public PreconditionOutcomeTable() {
+    exceptionSets = new ArrayList<>();
+    postConditions = new ArrayList<>();
+  }
+
+  /**
+   * Adds the outcome of checking the conditions of a specification.
+   *
+   * @param preconditionsSatisfied boolean value indicating whether all preconditions are satisfied
+   * @param throwsClauses set of exception type-comment pairs for exceptions expected in post-state
+   * @param postCondition post-condition that must be true in post-state if no exception is thrown,
+   *     null if none
+   */
+  void add(
+      boolean preconditionsSatisfied,
+      Set<ThrowsClause> throwsClauses,
+      PostCondition postCondition) {
+    isEmpty = false;
```

Comment:
A method call can set this field even if the method call has no other effect (if the precondition is not satisfied).
So, what is the semantics of this field?  Or, are there undocumented requirements on the method formal parameters that guarantee that every call makes a change?